src/Plugin/Field/FieldFormatter/ImageFormatter.php

```diff
@@ -26,6 +31,39 @@
 class ImageFormatter extends DrupalImageFormatter implements InstantArticleFormatterInterface {
 
   /**
+   * Base settings.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, $label, $view_mode, array $third_party_settings, AccountInterface $current_user, EntityStorageInterface $image_style_storage, ConfigFactoryInterface $config_factory) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $label, $view_mode, $third_party_settings, $current_user, $image_style_storage);
+    $this->config = $config_factory->get('fb_instant_articles.settings');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
```

Comment:
Nope, nothing specific to FBIA. This class defines a Field Formatter plugin, and field formatter plugins support dependency injection via this create method. Here I'm overriding the parent implementation, which you can find in `\Drupal\image\Plugin\Field\FieldFormatter\ImageFormatter`. Most of the stuff here is satisfying that implementation so as not to break things. This is a common pattern you'll find throughout Drupal in plugins.